cmd/cartesi-rollups-cli/root/inspect/inspect.go

```diff
@@ -0,0 +1,59 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+package inspect
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/cartesi/rollups-node/pkg/inspectclient"
+	"github.com/spf13/cobra"
+)
+
+var Cmd = &cobra.Command{
+	Use:     "inspect",
+	Short:   "Calls inspect API",
+	Example: examples,
+	Run:     run,
+}
+
+const examples = `# Makes a request with "hi" encoded as hex:
+cartesi-rollups-cli inspect --payload 0x$(printf "hi" | xxd -p)`
+
+var (
+	hexPayload      string
+	inspectEndpoint string
+)
+
+func init() {
+	Cmd.Flags().StringVar(&hexPayload, "payload", "",
+		"input payload hex-encoded starting with 0x")
+
+	cobra.CheckErr(Cmd.MarkFlagRequired("payload"))
+
+	Cmd.Flags().StringVar(&inspectEndpoint, "inspect-endpoint", "http://0.0.0.0:10009/",
+		"address used to connect to graphql")
+}
+
+func run(cmd *cobra.Command, args []string) {
+	ctx := cmd.Context()
+	client, err := inspectclient.NewClient(inspectEndpoint)
+	cobra.CheckErr(err)
+
+	requestBody := strings.NewReader(hexPayload)
+
+	response, err := client.InspectPostWithBody(ctx, "application/octet-stream", requestBody)
```

Comment:
This seems wrong. The code should decode the hex string before sending it to the API. (e.g. using [hexutil.Decode](https://pkg.go.dev/github.com/ethereum/go-ethereum/common/hexutil#Decode)).

Reply:
I sent `0xfafa` to an echo app and got back `0x307866616661`
```
➜  rollups-node git:(f544384d) go run ./cmd/cartesi-rollups-cli inspect --inspect-endpoint=http://127.0.0.1:8080 --payload 0xfafa
{
    "exception_payload": "0x",
    "processed_input_count": 0,
    "reports": [
        {
            "payload": "0x307866616661"
        }
    ],
    "status": "Accepted"
}
```